src/labbook/components/runtime.tsx

```diff
@@ -0,0 +1,225 @@
+import React from "react";
+import deepmerge from "deepmerge";
+import Shutterbug from "shutterbug";
+import hash from "object-hash";
+import { debounce } from "ts-debounce";
+
+import { ThumbnailChooser, IThumbnailChooserProps } from "./thumbnail-chooser/thumbnail-chooser";
+import { Thumbnail, IThumbnailProps } from "./thumbnail-chooser/thumbnail";
+// import { PreviewPanel } from "./preview-panel";
+import { UploadImage } from "./upload-image";
+import { CommentField } from "./comment-field";
+import { v4 as uuidv4 } from "uuid";
+import { IAuthoredState, IInteractiveState, ILabbookEntry } from "./types";
+import { DrawingTool, drawingToolCanvasSelector  } from "../../drawing-tool/components/drawing-tool";
+import { TakeSnapshot } from "./take-snapshot";
+import { IInteractiveState as IDrawingToolInteractiveState} from "../../drawing-tool/components/types";
+
+import css from "./runtime.scss";
+
+export interface IProps {
+  authoredState: IAuthoredState;
+  interactiveState?: IInteractiveState;
+  setInteractiveState?: (updateFunc: (prevState: IInteractiveState | null) => IInteractiveState) => void;
+  report?: boolean;
+}
+
+// convert 1-26 to A-Z.
+const numberToAlpha = (value:number) => (value + 10).toString(26).toUpperCase();
+
+const generateItem = () => {
+  const id = uuidv4();
+  const item:ILabbookEntry = {
+    data: {answerType: "interactive_state"},
+    comment: "",
+    id
+  };
+  return item;
+};
+
+
+export const Runtime: React.FC<IProps> = ({ authoredState, interactiveState, setInteractiveState, report }) => {
+
+
+  const ensureSelected = (prev: Partial<IInteractiveState>) => {
+    const result = { ...prev };
+    if ((result?.entries?.length|| -1) < 0) {
```

Comment:
I'm a little confused by this - is this here to ensure there is at least one entry?  If so should the comparison be `<= 0` instead?

Reply:
Because entries can be `undefined` in the partial, TS wasn't happy unless I cast undefined to a number.

I will try to think of a less confusing way to do this coercion.